package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -345,53 +447,79 @@ def read_DATA_timestep(self, n_atoms, TS_class, TS_kwargs,
 
         return ts
 
-    def _parse_pos(self, datalines):
-        """Strip coordinate info into np array"""
-        pos = np.zeros((len(datalines), 3), dtype=np.float32)
+    def _parse_pos(self, datalines, header):
+        """Strip coordinate info into a np array
+
+        Arguments
+        ---------
+        datalines : list of str
+            list of strings (raw lines) from data file.
+        header: dict of str
+            A dict in which a key and value pair is a 'header' name and
+            its value in str foramt.
+
+        Returns
+        -------
+        pos: np.ndarray
+            array of (x,y,z) coordinates of atoms in a data file.
+        order : np.ndarray
+            array of integers (resulted from applying np.argsort on atom ids
+            that corrrectly maps the velocities into their corrresponding atom
+            ids.
+        """
+        n_atoms = len(datalines)
+        if n_atoms != header['atoms']:
+            raise ValueError(
+                "Number of lines "
+                f"'{len(datalines)}'"
+                " in 'Atoms' section does not match the number of atoms "
+                f"'{header['atoms']}'."
+            )
+        pos = np.zeros((n_atoms, 3), dtype=np.float32)
         # TODO: could maybe store this from topology parsing?
         # Or try to reach into Universe?
-        # but ugly because assumes lots of things, and Reader should be standalone
+        # but ugly since assumes many things, and Reader should be standalone
         ids = np.zeros(len(pos), dtype=np.int32)
-
-        if self.style_dict is None:
-            if len(datalines[0].split()) in (7, 10):
-                style_dict = {'id': 0, 'x': 4, 'y': 5, 'z': 6}
-            else:
-                style_dict = {'id': 0, 'x': 3, 'y': 4, 'z': 5}
-        else:
-            style_dict = self.style_dict
-    
         for i, line in enumerate(datalines):
             line = line.split()
-
-            ids[i] = line[style_dict['id']]
-
-            pos[i, :] = [line[style_dict['x']],
-                         line[style_dict['y']],
-                         line[style_dict['z']]]
+            ids[i] = line[self.style_dict['id']]
```

Comment:
Can the positions be given in `xs, ys, zs` etc / other coordinated formats? Worth considering.

Reply:
As far as I understood from [read_data](https://docs.lammps.org/read_data.html) documentation, the unscaled coordinate are only written to a DATA file. The other coordinate format are used in DUMP files and properly handle with the lammps_coordinate_convention*lammps_coordinate_convention* keyword in the [DumpReader](https://github.com/MDAnalysis/mdanalysis/blob/ec5da1de682322c436970252f03b5a24ea673113/package/MDAnalysis/coordinates/LAMMPS.py#L454) class.